File: code/scala-demo/src/main/scala/test/c4/Test4.scala
package test.c4

/**
 * @Author 0726
 * @ClassName Test4
 * @createTime 2021年10月29日 18:26
 */
object Test4 {
  def main(args: Array[String]): Unit = {
    val arr = Iterable( ("宝安区",15),("宝安区",10),("福田区",3),("福田区",10))
    val map: Map[String, Iterable[(String, Int)]] = arr.groupBy(_._1)
    //println(map)

    map.map(x=>{
      val totalTime = x._2.map(x=>x._2).sum
      val count = x._2.size
      (x._1,totalTime.toDouble/count)
    })
      .foreach(println)

    val add = List(("宝安区",15),("宝安区",10),("福田区",3),("福田区",10))
    println(add.map(_._2))
    println(add.map(_._2).sum)

    println(add.map(x=>{
      val a = x._2
      ("aaa",a)
    }))


  }
}

File: code/0726project/homework/src/day05/Test4.java
package day05;

public class Test4 {
    public static void main(String[] args) {
        char[] ccc = {'a','l','f','m','f','o','b','b','s','n'};
        int[] count = new int[26];  //count[]该数组目前二十六个值都默认为零
        for (char c : ccc) {
            count[c - 97]++;
        }
        for(int i=0; i<count.length; i++){
            if(count[i]!=0){
                System.out.println((char)(i+97) + "--" + count[i]);
            }
        }
    }
}
